apps/vaporgui/QRangeSliderTextCombo.cpp

```diff
@@ -31,21 +32,54 @@ void QRangeSliderTextCombo::SetRange(float min, float max)
     VAssert(_max >= _min);
     _min = min;
     _max = max;
+    
+    if (_allowCustomRange) {
+        min = -FLT_MAX;
+        max = FLT_MAX;
+    }
+    
     setValidator(_leftText, new QDoubleValidator(min, max, 100));
     setValidator(_rightText, new QDoubleValidator(min, max, 100));
     SetValue(_left, _right);
 }
 
 void QRangeSliderTextCombo::SetValue(float left, float right)
 {
-    left = std::max(_min, left);
-    right = std::min(_max, right);
+    if (left > right)
+        std::swap(left, right);
+    if (!_allowCustomRange) {
+        left = std::max(_min, left);
+        right = std::min(_max, right);
+    }
     _left = left;
     _right = right;
     setTextboxes(left, right);
     _slider->SetValue((left-_min)/(_max-_min), (right-_min)/(_max-_min));
 }
 
+void QRangeSliderTextCombo::AllowCustomRange()
+{
+    assert(!_allowCustomRange);
```

Comment:
Do you really want this assert? It would appear that calling AllowCustomRange() more than once will fail the assert. Is that what you want?

If an assert is needed use VAssert

Reply:
Yes. If it is done, nothing will break so we don't want to crash the release application but generally if it is called multiple times there is a mistake someone made so it lets them know.

Reply:
Fixed